lib/src/mt_instance.c

```diff
@@ -0,0 +1,152 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright(c) 2023 Intel Corporation
+ */
+
+#include "mt_instance.h"
+
+#ifndef WINDOWSENV
+
+#include <sys/un.h>
+
+#include "../manager/mtl_mproto.h"
+#include "mt_log.h"
+
+int mt_instance_put_lcore(struct mtl_main_impl* impl, unsigned int lcore_id) {
+  int ret;
+  int sock = impl->instance_fd;
+
+  mtl_message_t msg;
+  msg.header.magic = htonl(MTL_MANAGER_MAGIC);
+  msg.header.type = htonl(MTL_MSG_TYPE_PUT_LCORE);
+  msg.body.lcore_msg.lcore = htons(lcore_id);
+  msg.header.body_len = sizeof(msg.body.lcore_msg);
+
+  ret = send(sock, &msg, sizeof(msg), 0);
+  if (ret < 0) {
+    err("%s, send message fail\n", __func__);
+    return ret;
+  }
+
+  return 0;
+}
+
+int mt_instance_get_lcore(struct mtl_main_impl* impl, unsigned int lcore_id) {
+  int ret;
+  int sock = impl->instance_fd;
+
+  mtl_message_t msg;
+  msg.header.magic = htonl(MTL_MANAGER_MAGIC);
+  msg.header.type = htonl(MTL_MSG_TYPE_GET_LCORE);
+  msg.body.lcore_msg.lcore = htons(lcore_id);
+  msg.header.body_len = htonl(sizeof(mtl_lcore_message_t));
+
+  ret = send(sock, &msg, sizeof(mtl_message_t), 0);
+  if (ret < 0) {
+    err("%s, send message fail\n", __func__);
+    return ret;
+  }
+
+  ret = recv(sock, &msg, sizeof(mtl_message_t), 0);
+
+  if (ret < 0 || ntohl(msg.header.magic) != MTL_MANAGER_MAGIC ||
+      ntohl(msg.header.type) != MTL_MSG_TYPE_RESPONSE) {
+    err("%s, recv response fail\n", __func__);
+    return -EIO;
+  }
+
+  int response = msg.body.response_msg.response;
+
+  return -response;
```

Comment:
nit: can you leave one comment on why `-response`?

Reply:
Thanks, done.